src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/HttpApplication/HttpApplicationMiddleware.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using System.Web;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.ObjectPool;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters;
+
+internal class HttpApplicationMiddleware
+{
+    private readonly RequestDelegate _next;
+    private readonly ObjectPool<HttpApplication> _pool;
+
+    public HttpApplicationMiddleware(RequestDelegate next, ObjectPool<HttpApplication> pool)
+    {
+        _next = next;
+        _pool = pool;
+    }
+
+    public async Task InvokeAsync(HttpContextCore context)
+    {
+        var app = _pool.Get();
```

Comment:
Do we need to preserve the pooling behavior at all? Can't we just have an allocation per request here?

Reply:
It also will new up and register all the modules. This can be potentially expensive, so I went with a pool. If someone doesn't want to have a pool, they could set the PoolSize to 0 or create a new ObjectPool<> that doesn't save anything

Reply:
I’m asking if pooling is required to be correct.

Reply:
pooling is not required to be correct; there was no guarantee about an instance being reused.